### PR TITLE
add payment link

### PR DESCRIPTION
**Pull request by vasylpopovych6**

### CODE REVIEW

File: src/components/fileMenu/index.jsx [hunk 1]
```diff
@@ -57,10 +64,9 @@ export const FileMenu = () => {
     dispatch(setSelectedFile({}));
   };
 
-  const onDeleteClick = () => {
-    dispatch(handleDeleteFileModal(true));
-    dispatch(handleFileMenu(false));
-  };
+  useEffect(() => {
+    setPPVState(isPPVActivated);
```
Comment (fifa-master) [inline src/components/fileMenu/index.jsx:68]: no need of this state

File: src/components/ppvModal/PPVModal.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  const isOpen = useSelector(selectPaperViewModal);
+  const [isProccess, setIsProccess] = useState(false);
+  const [isEditMode, setEditMode] = useState(false);
+  const user = useSelector((state) => state.user.data);
+  const [state, setState] = useState({ view: 1, download: 0, description: '' });
+  const ppvPayment = useSelector(selectPaymenttByKey('pay_per_view'));
+  const file = useSelector(selecSelectedFile);
+
+  useEffect(() => {
+    if (file.share_file) {
```
Comment (fifa-master) [inline src/components/ppvModal/PPVModal.jsx:31]: looks like useless

File: src/pages/paidView/components/preview/imagePreview/index.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import CN from 'classnames';
+
+import { FullscreenBtn } from '../../fullscreenBtn';
+import { removeExtension } from '../../../../../utils/string';
+
+import styles from './styles.module.css';
+
+export const ImagePreview = ({ file, fileContent, allowPreview, fullscreen, onFullscreen }) => {
+  const canvasRef = useRef(null);
+  const isSVG = useMemo(() => fileContent.endsWith('</svg>'));
```
Comment (fifa-master) [inline src/pages/paidView/components/preview/imagePreview/index.jsx:11]: missing deplist

File: src/pages/paidView/components/preview/index.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { AudioPreview } from './audioPreview';
+import { VideoPreview } from './videoPreview';
+
+import { getPreviewFileType } from '../../../../utils/preview';
+
+export const Preview = ({ file, fileContent, allowPreview, fullscreen, onFullscreen }) => {
+  const [previewFileType, setPreviewFileType] = useState('');
+
+  useEffect(() => {
+    setPreviewFileType(getPreviewFileType(file, fileContent));
```
Comment (fifa-master) [inline src/pages/paidView/components/preview/index.jsx:13]: useless
usememo

File: src/pages/paidView/components/preview/videoPreview/index.jsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+
+  const videoSize = useMemo(() => {
+    if (dimension.width < dimension.height) {
+      return { width: 'auto', height: '100%' };
+    } else {
+      return { width: '100%', height: 'auto' };
+    }
+  }, [dimension.width, dimension.height]);
+
+  const formatDuration = (duration) => {
```
Comment (fifa-master) [inline src/pages/paidView/components/preview/videoPreview/index.jsx:112]: move it to utils

File: src/components/fileItem/payShare.jsx [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+      <DotsIcon />
+    </button>
+  );
+
+  return (
+    <>
+      {view === 'grid' ? (
+        <li className={style.fileSquare} id={file.id} onClick={onClickHandler}>
+          {FavButton}
+          {MenuButton}
```
Comment (fifa-master) [inline src/components/fileItem/payShare.jsx:99]: better to use a <MenuButton/>

File: src/pages/paidView/components/preview/audioPreview/index.jsx [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        toast.error('Sorry, something went wrong. Please try again later');
+      })
+  }, [file.slug]);
+
+  const onFinish = useCallback(() => {
+    setIsPlaying(false);
+    setProgress(0);
+  }, []);
+
+  const formatDuration = (duration) => {
```
[the remark this answers is not in the document]
Comment (vasylpopovych6) [reply src/pages/paidView/components/preview/audioPreview/index.jsx:92]: done

File: src/pages/paidView/components/preview/videoPreview/index.jsx (hunk 5 above)
Comment (vasylpopovych6) [reply src/pages/paidView/components/preview/videoPreview/index.jsx:112]: done

File: src/pages/paidView/components/preview/audioPreview/index.jsx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+      setProgress(progressValue);
+
+      if (!allowPreview && audioRef.current.currentTime >= PAUSE_THRESHOLD_SECONDS) {
+        audioRef.current.pause();
+        setIsPlaying(false);
+      }
+    };
+
+    audioRef.current.addEventListener('timeupdate', updateProgress);
+
```
Comment (fifa-master) [inline src/pages/paidView/components/preview/audioPreview/index.jsx:71]: use onTimeUpdate

File: src/effects/filesEffects.js [hunk 9]
```diff
@@ -200,6 +200,33 @@ export const getFilesEffect = async (page = 1, order = 'desc') => {
   return await axiosInstance.get(url).then((result) => result.data);
 };
 
+export const getPaidShareFilesEffect = async (page = 1) => {
+  const url = `${API_PATH}/share/file/list?page=${page}`;
+  return await axiosInstance.get(url).then((result) => result.data);
```
Comment (fifa-master) [inline src/effects/filesEffects.js:205]: use or async/await or than/catch
dont use both

File: src/components/ppvModal/PPVModal.jsx [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+const PPVModal = () => {
+  const dispatch = useDispatch();
+  const isOpen = useSelector(selectPaperViewModal);
+  const [isProccess, setIsProccess] = useState(false);
+  const user = useSelector((state) => state.user.data);
+  const [state, setState] = useState({ view: 1, download: 0, description: '' });
+  const ppvPayment = useSelector(selectPaymenttByKey('pay_per_view'));
+  const file = useSelector(selecSelectedFile);
+  const isPPVActivated = useMemo(() => !!file.share_file, [file]);
+
```
Comment (fifa-master) [inline src/components/ppvModal/PPVModal.jsx:29]: better to use `file?.share_file` in deps list

File: src/components/fileItem/payShare.jsx [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+  return (
+    <li
+      className={view === 'grid' ? style.fileSquare : style.fileList}
+      id={file.id}
+      onClick={onClickHandler}
+    >
+      {FavButton}
+      {MenuButton}
+      <div className={view === 'grid' ? style.previewWrapper : style.fileListPreview}>
```
Comment (fifa-master) [inline src/components/fileItem/payShare.jsx:99]: separate `view === 'grid'` to new variable for better reading

**Review by fifa-master — APPROVED** [review]

**Pull request merged**